Instalation procedure:
  cwd=$(pwd)
  cd usr/lib/one/sunstone/public
  patch -p1 < $pwd/VMsnapshot-5.4.11.patch
  grunt sass
  grunt requirejs
  systemct restart opennebula-sunstone

diff -urN public-5.4.11/app/tabs/vms-tab/panels/snapshots.js public/app/tabs/vms-tab/panels/snapshots.js
--- public-5.4.11/app/tabs/vms-tab/panels/snapshots.js	2018-04-19 19:06:12.000000000 +0300
+++ public/app/tabs/vms-tab/panels/snapshots.js	2018-04-20 12:46:49.071484677 +0300
@@ -127,6 +127,13 @@
             if (Config.isTabActionEnabled("vms-tab", "VM.snapshot_delete")) {
               actions += '<a href="VM.snapshot_delete" class="snapshot_delete" ><i class="fa fa-times"/>' + Locale.tr("Delete") + '</a>'
             }
+          } else if (that.element.STATE == OpenNebulaVM.STATES.POWEROFF && that.element.HISTORY_RECORDS.HISTORY[that.element.HISTORY_RECORDS.HISTORY.length - 1].TM_MAD == "storpool"){
+            if (Config.isTabActionEnabled("vms-tab", "VM.snapshot_revert")) {
+              actions += '<a href="VM.snapshot_revert" class="snapshot_revert" ><i class="fa fa-reply"/>' + Locale.tr("Revert") + '</a> &emsp;'
+            }
+            if (Config.isTabActionEnabled("vms-tab", "VM.snapshot_delete")) {
+              actions += '<a href="VM.snapshot_delete" class="snapshot_delete" ><i class="fa fa-times"/>' + Locale.tr("Delete") + '</a>'
+            }
           }
         }
 
